### PR TITLE
Add new sections to Azure Kit documentation

### PR DESCRIPTION
**Pull request by tamasmak**
Adding new sections to the Azure Kit documentation:

- TLS
- Autoscaling
- Multiple Kubernetes Environments

The PR is draft, there will be additional updates about Terraform variables and settings.

### CODE REVIEW

File: articles/tools/azure/autoscaling.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
+= Autoscaling
+
+There are three different types of autoscaling possible:
+
+- The Horizontal Pod Autoscaler
+- The Vertical Pod Autoscaler
+- The Cluster Autoscaler
+
+== Horizoncal Pod Autoscaler
```
Comment (mcollovati) [inline articles/tools/azure/autoscaling.adoc:9]: Typo
```suggestion
== Horizontal Pod Autoscaler
```

File: articles/tools/azure/autoscaling.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+The https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/)[Horizontal Pod Autoscaler] controls the scale of a Deployment and its ReplicaSet.
+It is implemented as a Kubernetes API resource and a controller and cannot be deployed with this Terraform.
```
Comment (mcollovati) [inline articles/tools/azure/autoscaling.adoc:12]: Is `this` needed in the sentence?
```suggestion
It is implemented as a Kubernetes API resource and a controller and cannot be deployed with Terraform.
```

File: articles/tools/azure/tls.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+= TLS
+
+To help using `letsencrypt` there is an option in Terraform variables to enable `letsencrypt` `certmanager` in the cluster.
+
+After `certmanager` is installed you still need to https://learn.microsoft.com/en-us/azure/aks/ingress-tls?tabs=azure-cli#create-a-ca-cluster-issuer[create the cluster issuer separately].
+
+After this the next steps are to configure your ingress to use these certificates. You can follow the same Microsoft https://learn.microsoft.com/en-us/azure/aks/ingress-tls?tabs=azure-cli#update-your-ingress-routes[guide] for this.
```
Comment (mcollovati) [inline articles/tools/azure/tls.adoc:7]:
```suggestion
After this, the next steps are to configure your ingress to use these certificates. You can follow the same Microsoft https://learn.microsoft.com/en-us/azure/aks/ingress-tls?tabs=azure-cli#update-your-ingress-routes[guide] for this.
```

File: articles/tools/azure/autoscaling.adoc (hunk 1 above)
Comment (tamasmak) [reply articles/tools/azure/autoscaling.adoc]: Fixed

File: articles/tools/azure/autoscaling.adoc (hunk 2 above)
Comment (tamasmak) [reply articles/tools/azure/autoscaling.adoc]: Fixed. Thanks

File: articles/tools/azure/tls.adoc (hunk 3 above)
Comment (tamasmak) [reply articles/tools/azure/tls.adoc]: Fixed

**Review by tarekoraby — APPROVED** [review]

**Pull request merged**